core/src/main/scala/org/broadinstitute/dsde/rawls/user/UserService.scala

```diff
@@ -651,60 +651,126 @@ class UserService(protected val ctx: RawlsRequestContext,
       }
+    } yield {}
 
+  def addUserToBillingProjectV2(projectName: RawlsBillingProjectName,
+                                projectAccessUpdate: ProjectAccessUpdate
+  ): Future[Unit] =
+    requireProjectAction(projectName, SamBillingProjectActions.alterPolicies) {
       for {
-        _ <- Future.traverse(policies) { policy =>
-          samDAO
-            .addUserToPolicy(SamResourceTypeNames.billingProject,
-                             projectName.value,
-                             policy,
-                             projectAccessUpdate.email,
-                             ctx.userInfo
+        billingProjectId <- getBillingProfileId(projectName)
+        policies = billingProjectId match {
+          case None => getLegacyBillingPolicies(projectAccessUpdate.role)
+          case Some(billingProjectId) =>
+            billingProfileManagerDAO.addProfilePolicyMember(
+              UUID.fromString(billingProjectId),
+              projectAccessUpdate.role,
+              projectAccessUpdate.email,
+              ctx
             )
-            .recoverWith { case regrets: Throwable =>
-              if (policy == SamBillingProjectPolicyNames.canComputeUser) {
-                logger.info(
-                  s"error adding user to canComputeUser policy for $projectName likely because it is a v2 billing project which does not have a canComputeUser policy. regrets: ${regrets.getMessage}"
-                )
-                Future.successful(())
-              } else {
-                Future.failed(regrets)
-              }
-            }
+            Seq(getV2BillingPolicy(projectAccessUpdate.role))
         }
+        _ <- addUserToBillingProjectInner(projectName, projectAccessUpdate, policies)
       } yield {}
     }
 
+  def getBillingProfileId(projectName: RawlsBillingProjectName): Future[Option[String]] =
+    dataSource.inTransaction { dataAccess =>
+      dataAccess.rawlsBillingProjectQuery.load(projectName).map { billingProjectOpt =>
+        billingProjectOpt
+          .getOrElse(
+            throw new RawlsException(s"Billing Project ${projectName.value} does not exist in Rawls database")
+          )
+          .billingProfileId
+      }
+    }
+
   def removeUserFromBillingProject(projectName: RawlsBillingProjectName,
                                    projectAccessUpdate: ProjectAccessUpdate
   ): Future[Unit] =
     requireProjectAction(projectName, SamBillingProjectActions.alterPolicies) {
-      val policy = projectAccessUpdate.role match {
-        case ProjectRoles.Owner => SamBillingProjectPolicyNames.owner
-        case ProjectRoles.User  => SamBillingProjectPolicyNames.workspaceCreator
-      }
+      removeUserFromBillingProjectInner(projectName, projectAccessUpdate)
+    }
+
+  private def removeUserFromBillingProjectInner(projectName: RawlsBillingProjectName,
+                                                projectAccessUpdate: ProjectAccessUpdate
+  ): Future[Unit] =
+    for {
+      _ <- samDAO
```

Comment:
If you aren't doing anything with the results, and it's a single async operation, the for construct feels a bit gratuitous. 
If you remove the for, and do nothing else, it's exactly the same functionality.

Reply:
Thanks, will do!